runner/lib/main.js

```diff
@@ -725,20 +727,32 @@ const main = async (progName, rawArgs, powers) => {
 
       const notifier = {
         currentCount: 0,
-        /** @param {number} count */
-        updateActive(count) {
-          notifier.currentCount = count;
-          if (!count && notifier.idleCallback) {
-            notifier.idleCallback();
+        totalCycles: 0,
+        /**
+         * @param {string} _task
+         * @param {number} _seq
+         */
+        start(_task, _seq) {
+          notifier.currentCount += 1;
+          notifier.totalCycles += 1;
+          if (notifier.totalCycles >= cycleCount) {
+            stopLoadgenKit.resolve();
           }
         },
         /**
          * @param {string} task
          * @param {number} seq
+         * @param {boolean} success
          */
-        taskFailure(task, seq) {
-          loadgenTaskFailed = new Error(`Loadgen ${task} task ${seq} failed`);
-          stopLoadgenKit.resolve();
+        finish(task, seq, success) {
+          const count = notifier.currentCount - 1;
+          notifier.currentCount = count;
+          if (!success) {
+            loadgenTaskFailed = new Error(`Loadgen ${task} task ${seq} failed`);
+            stopLoadgenKit.resolve();
+          } else if (!count && notifier.idleCallback) {
+            notifier.idleCallback();
+          }
```

Comment:
this reads more clearly:
```suggestion
          if (success) {
            if (count === 0 && notifier.idleCallback)
              notifier.idleCallback();
            }
          } else {
            loadgenTaskFailed = new Error(`Loadgen ${task} task ${seq} failed`);
            stopLoadgenKit.resolve();
          } 
```
though since `notifier` knows its count, maybe it should be responsible for conditioning on it?

Reply:
Yeah the whole `notifier` thing is a little hacky